src/modules/dashboard/components/EditDomainDialog/EditDomainDialog.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useCallback } from 'react';
+import { FormikProps } from 'formik';
+import * as yup from 'yup';
+import { useHistory } from 'react-router-dom';
+import { ROOT_DOMAIN_ID } from '@colony/colony-js';
+
+import Dialog, { DialogProps } from '~core/Dialog';
+import { ActionForm } from '~core/Fields';
+
+import { Colony } from '~data/index';
+import { ActionTypes } from '~redux/index';
+import { WizardDialogType } from '~utils/hooks';
+import { pipe, withMeta, mapPayload } from '~utils/actions';
+
+import DialogForm from './EditDomainDialogForm';
+import { Color } from '~core/ColorTag';
+
+export interface FormValues {
+  domainId: string;
+  domainName: string;
+  domainColor?: Color;
+  domainPurpose?: string;
+  annotationMessage?: string;
+}
+
+interface CustomWizardDialogProps {
+  prevStep?: string;
+  colony: Colony;
+  selectedDomainId?: string;
+}
+
+type Props = DialogProps &
+  Partial<WizardDialogType<object>> &
+  CustomWizardDialogProps;
+
+const displayName = 'dashboard.EditDomainDialog';
+
+const EditDomainDialog = ({
+  callStep,
+  prevStep,
+  cancel,
+  close,
+  colony,
+  colony: { colonyAddress, colonyName, domains },
+  selectedDomainId,
+}: Props) => {
+  const history = useHistory();
+
+  const validationSchema = yup.object().shape({
+    domainName: yup.string(),
```

Comment:
I think we should restrict these to the value we set in the input's `maxLength` both for `domainName` and `domainPurpose`